packages/react-components/react-progress/src/components/ProgressBar/useProgressBar.tsx

```diff
@@ -13,14 +14,16 @@ import type { ProgressBarProps, ProgressBarState } from './ProgressBar.types';
  */
 export const useProgressBar_unstable = (props: ProgressBarProps, ref: React.Ref<HTMLElement>): ProgressBarState => {
   // Props
-  const { color = 'brand', max = 1.0, shape = 'rounded', thickness = 'medium', value } = props;
+  const { color = 'brand', max = 1, shape = 'rounded', thickness = 'medium', value } = props;
+  const internalMax = clampMax(max);
+  const internalValue = clampValue(value, internalMax);
```

Comment:
(nit) This works ok, but a future change might end up using `max` instead of `internalMax` by mistake. Rather than having two separate variables, you can define `max` and `value` directly:
```suggestion
  const { color = 'brand', shape = 'rounded', thickness = 'medium' } = props;
  const max = clampMax(props.max ?? 1);
  const value = clampValue(props.value, max);
```